### PR TITLE
i18n-calypso: add the isRtl functionality and remove components/rtl

### PR DESCRIPTION
**Pull request by jsnajdr**
Currently the helpers in `components/rtl` use `lib/i18n-utils` to get the current locale's RTL flag, using the info from the [_client/languages.js_ module](https://github.com/Automattic/wp-calypso/blob/master/client/languages.js). But that's Calypso-specific and code that uses only `i18n-calypso` and `@automattic/components` doesn't have access to that.

Fortunately, the RTL flag is encoded in our translation files, too:
- Core translations include a special translated string `"ltr"`. If it translates to `"rtl"`, we have a RTL language!
- Calypso translations include a `[ '' ][ 'momentjs_locale' ][ 'textDirection' ]` property that has either the `ltr` or `rtl` value

I moved the `components/rtl` functionality to `i18n-calypso` and retrieve the RTL flag from the places mentioned above.

That makes the `i18n-calypso` dependency the only thing you need for your components to be localized.

**How to test:**
Verify that the RTL flag returned by `i18n.isRtl()` and the `withRtl` and `useRtl` wrappers is correct. One way to do that is to follow the testing steps from #36854 and test the Masterbar Publish popover.

### CODE REVIEW

File: packages/i18n-calypso/src/i18n.js [hunk 1]
```diff
@@ -291,6 +292,12 @@ I18N.prototype.setLocale = function( localeData ) {
 
 	this.state.localeSlug = this.state.locale[ '' ].localeSlug;
 
+	// extract the `textDirection` info (LTR or RTL) from either:
+	// - the translation for the special string "ltr" (standard in Core, not present in Calypso)
+	// - or the `momentjs_locale.textDirection` property present in Calypso translation files
+	this.state.textDirection =
+		this.state.locale.ltr || this.state.locale[ '' ]?.momentjs_locale?.textDirection;
```
Comment (akirk) [inline packages/i18n-calypso/src/i18n.js]: We actually use a context for this: https://translate.wordpress.com/projects/wpcom/-all-translated/41881/ resp.https://translate.wordpress.org/projects/wp/dev/he/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=40305&filters%5Btranslation_id%5D=1516997 
```suggestion
		this.state.locale[ 'text direction\u0004ltr' ] || this.state.locale[ '' ]?.momentjs_locale?.textDirection;
```

**Review by akirk — CHANGES_REQUESTED** [review]
Nice work, just the check for the `rtl` is not quite right yet. I suggested the necessary code change and am deploying our new language files that include the translation, so that we can test this with RTL languages (although it falls back to the already existing textDirection which might make this a little harder to test).

**Comment by akirk** [comment]
http://widgets.wp.com/languages/calypso/he-v1.1.json?v=46661
```
"text direction\u0004ltr":["rtl"]
```

File: packages/i18n-calypso/src/i18n.js (hunk 1 above)
Comment (jsnajdr) [reply packages/i18n-calypso/src/i18n.js]: Thanks @akirk! I didn't know the context is part of the translation key and that the value is an array. Fixed (I added an optional `[ 0 ]` access to your suggestion) and ready for another review!

**Comment by jsnajdr** [comment]
@akirk This is ready for final review. The linter errors reported by CircleCI are all by the picky JSDoc rule that doesn't like format of many doc comments.

**Comment by blowery** [comment]
@jsnajdr looks like the linter failed to run entirely? weird.

**Review by akirk — APPROVED** [review]
I tested this (I removed this line to make sure the translation data is being used: https://github.com/Automattic/wp-calypso/pull/37593/files#diff-8272e039681c191b99c69876d2af557cR299) and it worked, so good to go from me on the functional side. I didn't check the failing CI.

**Comment by jsnajdr** [comment]
> looks like the linter failed to run entirely? weird.

@blowery I see 41 reported violations of the jsdoc rule, and some weird error about failing to run the lint 😕 Anyway, I need to rebase and resolve conflicts, let's see what a fresh run says...

**Comment by jsnajdr** [comment]
Linter still reports 41 jsdoc failures, as on the previous run. Merging now.

**Pull request merged**